src/main/java/frc/robot/subsystems/ColorPanelSubsystem.java

```diff
@@ -0,0 +1,148 @@
+            colorString = "BLUE";
+        } else if (match.color == kRedTarget) {
+            colorString = "RED";
+        } else if (match.color == kGreenTarget) {
+            colorString = "GREEN";
+        } else if (match.color == kYellowTarget) {
+            colorString = "YELLOW";
+        } else {
+            colorString = "UNKNOWN";
+        }
+
+        /**
+         * Open Smart Dashboard 
+         * or Shuffleboard to see the color detected by the 
+         * sensor.
+         */
+        SmartDashboard.putNumber("RED", detectedColor.red);
+        SmartDashboard.putNumber("GREEN", detectedColor.green);
+        SmartDashboard.putNumber("BLUE", detectedColor.blue);
+        SmartDashboard.putNumber("Confidence", match.confidence);
+        SmartDashboard.putString("Detected Color", colorString);
+     
+        
+    }
+    public String returnColor() {
+      /**
+       * The method GetColor() returns a normalized color value from the sensor and can be
+       * useful if outputting the color to an RGB LED or similar. To
+       * read the raw color, use GetRawColor().
+       * 
+       * The color sensor works best when within a few inches from an object in
+       * well lit conditions (the built in LED is a big help here!). The farther
+       * an object is the more light from the surroundings will bleed into the 
+       * measurements and make it difficult to accurately determine its color.
+       */
+      Color detectedColor = colorSensor.getColor();
+
+      /**
+       * Run the color match algorithm on our detected color
+       */
+      String colorString;
+      ColorMatchResult match =  colorMatcher.matchClosestColor(detectedColor);
+
+      if (match.color == kBlueTarget) {
+          colorString = "BLUE";
+      } else if (match.color == kRedTarget) {
+          colorString = "RED";
+      } else if (match.color == kGreenTarget) {
+          colorString = "GREEN";
+      } else if (match.color == kYellowTarget) {
+          colorString = "YELLOW";
+      } else {
+          colorString = "UNKNOWN";
+      }
+
+      /**
+       * Open Smart Dashboard 
+       * or Shuffleboard to see the color detected by the 
+       * sensor.
+       */
+   
+      return colorString;
+      
+  }
+
+
+    public void fieldColor() {
+        String gameData;
+gameData = DriverStation.getInstance().getGameSpecificMessage();
+if(gameData.length() > 0)
+{
+  switch (gameData.charAt(0))
+  {
+    case 'B' :
+      SmartDashboard.putBoolean("Blue",true);
+      break;
+    case 'G' :
+    SmartDashboard.putBoolean("Green",true);
+      break;
+    case 'R' :
+    SmartDashboard.putBoolean("Red",true);
+      break;
+    case 'Y' :
+    SmartDashboard.putBoolean("Yellow",true);
+      break;
+    default :
+    SmartDashboard.putBoolean("Color",false);
+      break;
+  }
+} else {
+  SmartDashboard.putBoolean("Color",false);
+}
+    }
+    
+    
+    public void periodic (){
+        detectColor();
```

Comment:
(1) these methods do nothing.
(2) we should include an override tag, if the intention is to override a superclass method.
(3) we can probably eliminate these, and just have getters that return the various colors. 
These can then be called from subsystemManager

If we do leave a periodic method, it would be a good place to log data to the smart dashboard